prime-router/src/main/kotlin/azure/WorkflowEngine.kt

```diff
@@ -308,23 +308,35 @@ class WorkflowEngine(
             )
             val ids = tasks.map { it.reportId }
             val reportFiles = ids
-                .map { db.fetchReportFile(it, org = null, txn = txn) }
+                .mapNotNull {
+                    try {
+                        db.fetchReportFile(it, org = null, txn = txn)
+                    } catch (e: Exception) {
+                        println(e.printStackTrace())
+                        // sanityCheckReport below will log the problem in better detail.
+                        // note that we are logging but ignoring this error, so that it doesn't poison the entire batch
+                        null // id not found. Can occur if errors in ReportFunction fail to write to REPORT_FILE
```

Comment:
This is the key fix to #2180.   Skip over reports that are in the TASK table but missing from the REPORT_FILE table.

Reply:
How could a task for a report end up in the database without there being a report in the REPORT_FILE table? Is there a case to resolve that issue?

Reply:
Yes, see #2185